### DISCUSSION TITLE
Where does the function "init(" come from in /res/wrap/wrap.jl:31

### DISCUSSION THREAD

**Discussion by Marc-3d** [JuliaGPU/CUDA.jl#1608] [Q&A] [answered]
Hi CUDA.jl people, 

I am looking to use CUDNN to do cross correlations for my work. I have been looking at the source code of the CUDNN wrapper because I want to understand what goes on behind the scenes when I call CUDA.CUDNN.cudnnConvolutionForward(). 

Fast forward a couple days and I am reading through wrap.jl, in CUDA.jl/res/wrap/. I understand what wrappers are and what "wrap.jl" does, but I am can't figure out what module exports the "init" function in[ line 31](https://github.com/JuliaGPU/CUDA.jl/blob/0c5bd736f91877c3dfac1d08af3448bd08733d00/res/wrap/wrap.jl#L31). Could someone shed some light on this?

 I didn't want to open an Issue for this... so I decided to post he question here. 

Thanks in advance, 
Marc

**Comment by maleadt** [comment, marked as answer]
From the old version of Clang.jl we're using: https://github.com/JuliaInterop/Clang.jl/blob/v0.13.0/src/Clang.jl#L65

If you want to improve wrap.jl, a good first step would be to upgrade it to the latest version of Clang.jl
